### DISCUSSION TITLE
[Feature]: Use deployment feature of GitHub

### DISCUSSION THREAD

**Discussion by GunniBusch** [coollabsio/coolify#2931] [Feature Requests]
### Description

It would be helpful, to allow the GitHub App, to use the GitHub deployment functions. this would allow, to restrict deployments and to use deployments as pr tests before merging.
Here are some infos:
https://docs.github.com/en/rest/guides/delivering-deployments?apiVersion=2022-11-28
https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#about-deployments

### Minimal Reproduction (if possible, example repository)

-

### Version

*

**Comment by andrasbacsai** [comment]
Thank you for the suggestions. Features should be added to the `Discussion` section, so I will convert this one to it.
